game/src/Game/GameSession/BillboardManager.gd

```diff
@@ -0,0 +1,191 @@
+func set_capitals() -> void:
+	var positions = MapItemSingleton.get_capital_positions()
+	var capital_positions : PackedVector3Array = to_map_coords(positions)
+
+	multimesh.visible_instance_count = capitals_begin_index + capital_positions.size()
+	for i in capital_positions.size():
+		var capital_index = capitals_begin_index + i
+		multimesh.set_instance_transform(capital_index, Transform3D(Basis(), 
+			capital_positions[i]
+		))
+		
+		var im = billboard_names["capital"]
+		#capital image, frame=1 ,2x unused
+		#frame=1 because frame=0 would cause capitals not to show
+		#and as an index its fine, since the shader UVs will wrap around
+		# 1.0 to 2.0 --> 0.0 to 1.0 so the capital image is preserved
+		multimesh.set_instance_custom_data(capital_index,Color(
+			im,1,0,0
+		))
+
+# should provinces display RGO, crime, ..., or no billboard
+func set_province_billboards(display : ProvinceBillboards = ProvinceBillboards.INVISIBLE) -> void:
+	var image_index = 0
+	var icons:PackedByteArray = PackedByteArray()
+	icons.resize(capitals_begin_index)
+	icons.fill(0) #by default, display nothing (invisible)
+	match display:
+		ProvinceBillboards.RGO:
+			image_index = billboard_names["tradegoods"]
+			icons = MapItemSingleton.get_rgo_icons()
+			current_province_billboard = display
+		ProvinceBillboards.CRIME:
+			image_index = billboard_names["crimes"]
+			icons = MapItemSingleton.get_crime_icons()
+			current_province_billboard = display
+		ProvinceBillboards.NATIONAL_FOCUS:
+			image_index = billboard_names["national_focus"]
+			icons = MapItemSingleton.get_national_focus_icons()
+			current_province_billboard = display
+		ProvinceBillboards.NONE:
+			current_province_billboard = display
+		_: #display nothing, but keep the current billboard setting
+			pass
+
+	# for every billboard before the capitals (ie. the province billboards)
+	for i in capitals_begin_index:
+		multimesh.set_instance_custom_data(i,Color(
+			image_index,icons[i],0,0
+		))
+
+func _on_game_state_changed() -> void:
+	set_province_billboards(current_province_billboard)
+	set_capitals()
+
+# There are essentially 3 visibility states we can be in
+# 1: parchment view -> no billboards visible
+# 2: not parchment nor detail -> only capitals visible
+# 3: detail map -> province and capital billboards visible
+# So set_visible here is essentially to toggle the visibility of capitals
+
+func detailed_map(visible : bool) -> void:
+	if visible:
+		set_visible(true)
+		set_province_billboards(current_province_billboard)
+	else:
+		set_visible(true)
+		set_province_billboards()
+
+func parchment_view(is_parchment : bool) -> void:
+	if is_parchment:
+		set_visible(false)
+	else:
+		detailed_map(false)
+
+func _on_map_mode_changed(map_mode : int) -> void:
+	match map_mode:
+		5,6,10: # Infrastructure, Colonial, RGO output
+			set_province_billboards(ProvinceBillboards.RGO)
+		2: #Revolt Risk
+			set_province_billboards(ProvinceBillboards.CRIME)
+		9: #National Focus
```

Comment:
Maybe create an enum of mapmode indices, even just for use here, to make it cleaner/avoid "magic numbers"